mpc85xx: tl-wdr4900: use NVMEM for WAN

neheb opened the pull request:
Now that the ports are defined in DTS, this is trivial to do.

Avoids userspace handling.

ping @CHKDSK88

neheb committed:
mpc85xx: tl-wdr4900: use NVMEM for WAN

Now that the ports are defined in DTS, this is trivial to do.

Avoids userspace handling.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16125
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb committed:
mpc85xx: use NVMEM for wifi

Userspace handling is deprecated. No need for any of this.

Calibration size was adjusted to 440, which is the standard value for
ath9k radios.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16125
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb commented:
Noticed the wifi was handled in userspace. Adjusted that as well.

I'm not sure if those wifi nodes need to be swapped.

Noltari commented on target/linux/mpc85xx/files/arch/powerpc/boot/dts/tl-wdr4900-v1.dts:
```diff
@@ -148,6 +164,10 @@
+
+						nvmem-cells = <&macaddr_uboot_4fc00 1>;
+						nvmem-cell-names = "mac-address";
+
```
Empty line

neheb replied:
Done.

neheb commented:
From looking at a bootlog, it seems the pcie nodes are correct.

CHKDSK88 commented:
Wifi error:

```[    0.000000] Memory CAM mapping: 16/16/16/16/64 Mb, residual: 0Mb
[    0.000000] Activating Kernel Userspace Access Protection
[    0.000000] Activating Kernel Userspace Execution Prevention
[    0.000000] Linux version 6.6.45 (x@y) (powerpc-openwrt-linux-musl-gcc (OpenWrt GCC 13.3.0 r0+27128-9cb11a9715c4) 13.3.0, GNU ld (GNU Binutils) 2.42) #0 Mon Aug 19 06:12:53 2024
[    0.000000] Hardware name: TP-Link TL-WDR4900 v1 e500v2 0x80212151 Freescale P1014
[    0.000000] Found legacy serial port 0 for /soc@ffe00000/serial@4500
[    0.000000]   mem=ffe04500, taddr=ffe04500, irq=0, clk=399999996, speed=0
[    0.000000] -----------------------------------------------------
[    0.000000] phys_mem_size     = 0x8000000
[    0.000000] dcache_bsize      = 0x20
[    0.000000] icache_bsize      = 0x20
[    0.000000] cpu_features      = 0x0000000010010108
[    0.000000]   possible        = 0x0000000010010108
[    0.000000]   always          = 0x0000000010010108
[    0.000000] cpu_user_features = 0x84e08000 0x08000000
[    0.000000] mmu_features      = 0x00020210
[    0.000000] -----------------------------------------------------
[    0.000000] TL-WDR4900 v1 board from TP-Link
[    0.000000] barrier-nospec: using isync; sync as speculation barrier
[    0.000000] barrier-nospec: patched 0 locations
[    0.000000] Top of RAM: 0x8000000, Total RAM: 0x8000000
[    0.000000] Memory hole size: 0MB
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] MMU: Allocated 1088 bytes of context maps for 255 contexts
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Kernel command line: console=ttyS0,115200
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Kernel virtual memory layout:
[    0.000000]   * 0xffbdf000..0xfffff000  : fixmap
[    0.000000]   * 0xc9000000..0xffbdf000  : vmalloc & ioremap
[    0.000000] Memory: 112228K/131072K available (7676K kernel code, 668K rwdata, 8708K rodata, 192K init, 223K bss, 18844K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 512, nr_irqs: 512, preallocated irqs: 16
[    0.000000] mpic: Resetting
[    0.000000] mpic: Setting up MPIC " OpenPIC  " version 1.2 at ffe40000, max 1 CPUs
[    0.000000] mpic: ISU size: 256, shift: 8, mask: ff
[    0.000000] mpic: Initializing for 256 sources
[    0.000000] time_init: decrementer frequency = 49.999999 MHz
[    0.000000] time_init: processor frequency   = 799.999992 MHz
[    0.000006] clocksource: timebase: mask: 0xffffffffffffffff max_cycles: 0xb88126fc8, max_idle_ns: 440795204701 ns
[    0.000016] clocksource: timebase mult[14000007] shift[24] registered
[    0.000031] clockevent: decrementer mult[cccccc9] shift[32] cpu[0]
[    0.000091] pid_max: default: 32768 minimum: 301
[    0.006742] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.006758] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.012339] RCU Tasks Trace: Setting shift to 0 and lim to 1 rcu_task_cb_adjust=1.
[    0.015215] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.015240] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.018516] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.023682] Found FSL PCI host bridge at 0x00000000ffe09000. Firmware bus number: 0->255
[    0.023698] PCI host bridge /pcie@ffe09000 (primary) ranges:
[    0.023719]  MEM 0x00000000a0000000..0x00000000bfffffff -> 0x00000000a0000000
[    0.023729]   IO 0x00000000ffc10000..0x00000000ffc1ffff -> 0x0000000000000000
[    0.023755] /pcie@ffe09000: PCICSRBAR @ 0xfff00000
[    0.023765] setup_pci_atmu: end of DRAM 8000000
[    0.024205] Found FSL PCI host bridge at 0x00000000ffe0a000. Firmware bus number: 0->255
[    0.024216] PCI host bridge /pcie@ffe0a000  ranges:
[    0.024234]  MEM 0x0000000080000000..0x000000009fffffff -> 0x0000000080000000
[    0.024244]   IO 0x00000000ffc00000..0x00000000ffc0ffff -> 0x0000000000000000
[    0.024264] /pcie@ffe0a000: PCICSRBAR @ 0xfff00000
[    0.024273] setup_pci_atmu: end of DRAM 8000000
[    0.026553] PCI: Probing PCI hardware
[    0.026679] fsl-pci ffe09000.pcie: PCI host bridge to bus 9000:00
[    0.026691] pci_bus 9000:00: root bus resource [io  0x0000-0xffff]
[    0.026702] pci_bus 9000:00: root bus resource [mem 0xa0000000-0xbfffffff]
[    0.026713] pci_bus 9000:00: root bus resource [bus 00-ff]
[    0.026724] pci_bus 9000:00: busn_res: [bus 00-ff] end is updated to ff
[    0.026768] pci 9000:00:00.0: [1957:012b] type 01 class 0x060400
[    0.026788] pci 9000:00:00.0: reg 0x10: [mem 0xfff00000-0xffffffff]
[    0.026855] pci 9000:00:00.0: supports D1 D2
[    0.026863] pci 9000:00:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    0.027976] pci 9000:01:00.0: [168c:0033] type 00 class 0x028000
[    0.028006] pci 9000:01:00.0: reg 0x10: [mem 0xa0000000-0xa001ffff 64bit]
[    0.028048] pci 9000:01:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    0.028136] pci 9000:01:00.0: supports D1
[    0.028143] pci 9000:01:00.0: PME# supported from D0 D1 D3hot
[    0.050141] pci 9000:00:00.0: PCI bridge to [bus 01-ff]
[    0.050165] pci 9000:00:00.0:   bridge window [mem 0xa0000000-0xa00fffff]
[    0.050187] pci_bus 9000:01: busn_res: [bus 01-ff] end is updated to 01
[    0.050202] pci_bus 9000:00: busn_res: [bus 00-ff] end is updated to 01
[    0.050474] fsl-pci ffe0a000.pcie: PCI host bridge to bus a000:02
[    0.050488] pci_bus a000:02: root bus resource [io  0x20000-0x2ffff] (bus address [0x0000-0xffff])
[    0.050498] pci_bus a000:02: root bus resource [mem 0x80000000-0x9fffffff]
[    0.050509] pci_bus a000:02: root bus resource [bus 02-ff]
[    0.050518] pci_bus a000:02: busn_res: [bus 02-ff] end is updated to ff
[    0.050577] pci a000:02:00.0: [1957:012b] type 01 class 0x060400
[    0.050597] pci a000:02:00.0: reg 0x10: [mem 0xfff00000-0xffffffff]
[    0.050658] pci a000:02:00.0: supports D1 D2
[    0.050665] pci a000:02:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    0.051707] pci a000:02:00.0: Primary bus is hard wired to 0
[    0.051718] pci a000:02:00.0: bridge configuration invalid ([bus 01-01]), reconfiguring
[    0.051835] pci a000:03:00.0: AR9381 fixup applied
[    0.051846] pci a000:03:00.0: [168c:0030] type 00 class 0x028000
[    0.051873] pci a000:03:00.0: reg 0x10: [mem 0x80000000-0x8001ffff 64bit]
[    0.051915] pci a000:03:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    0.052003] pci a000:03:00.0: supports D1
[    0.052010] pci a000:03:00.0: PME# supported from D0 D1 D3hot
[    0.080151] pci a000:02:00.0: PCI bridge to [bus 03-ff]
[    0.080170] pci a000:02:00.0:   bridge window [mem 0x80000000-0x800fffff]
[    0.080188] pci_bus a000:03: busn_res: [bus 03-ff] end is updated to 03
[    0.080201] pci_bus a000:02: busn_res: [bus 02-ff] end is updated to 03
[    0.080314] PCI: Cannot allocate resource region 0 of device 9000:00:00.0, will remap
[    0.080329] PCI: Cannot allocate resource region 0 of device a000:02:00.0, will remap
[    0.080345] PCI 9000:00 Cannot reserve Legacy IO [io  0x0000-0x0fff]
[    0.080355] PCI a000:02 Cannot reserve Legacy IO [io  0x20000-0x20fff]
[    0.080379] pci 9000:00:00.0: BAR 0: no space for [mem size 0x00100000]
[    0.080390] pci 9000:00:00.0: BAR 0: failed to assign [mem size 0x00100000]
[    0.080404] pci 9000:01:00.0: BAR 6: assigned [mem 0xa0020000-0xa002ffff pref]
[    0.080416] pci 9000:00:00.0: PCI bridge to [bus 01]
[    0.080425] pci 9000:00:00.0:   bridge window [io  0x0000-0xffff]
[    0.080435] pci 9000:00:00.0:   bridge window [mem 0xa0000000-0xbfffffff]
[    0.080446] pci_bus 9000:00: Some PCI device resources are unassigned, try booting with pci=realloc
[    0.080453] pci_bus 9000:00: resource 4 [io  0x0000-0xffff]
[    0.080462] pci_bus 9000:00: resource 5 [mem 0xa0000000-0xbfffffff]
[    0.080472] pci_bus 9000:01: resource 0 [io  0x0000-0xffff]
[    0.080481] pci_bus 9000:01: resource 1 [mem 0xa0000000-0xbfffffff]
[    0.080494] pci a000:02:00.0: BAR 0: no space for [mem size 0x00100000]
[    0.080504] pci a000:02:00.0: BAR 0: failed to assign [mem size 0x00100000]
[    0.080515] pci a000:03:00.0: BAR 6: assigned [mem 0x80020000-0x8002ffff pref]
[    0.080526] pci a000:02:00.0: PCI bridge to [bus 03]
[    0.080535] pci a000:02:00.0:   bridge window [io  0x20000-0x2ffff]
[    0.080544] pci a000:02:00.0:   bridge window [mem 0x80000000-0x9fffffff]
[    0.080554] pci_bus a000:02: Some PCI device resources are unassigned, try booting with pci=realloc
[    0.080560] pci_bus a000:02: resource 4 [io  0x20000-0x2ffff]
[    0.080570] pci_bus a000:02: resource 5 [mem 0x80000000-0x9fffffff]
[    0.080579] pci_bus a000:03: resource 0 [io  0x20000-0x2ffff]
[    0.080588] pci_bus a000:03: resource 1 [mem 0x80000000-0x9fffffff]
[    0.086815] EDAC MC: Ver: 3.0.0
[    0.088499] clocksource: Switched to clocksource timebase
[    0.095753] NET: Registered PF_INET protocol family
[    0.095917] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.096386] tcp_listen_portaddr_hash hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.096413] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.096426] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.096443] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.096470] TCP: Hash tables configured (established 1024 bind 1024)
[    0.096571] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.096598] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.097084] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.097168] PCI: CLS 32 bytes, default 32
[    0.100392] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.101411] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.101419] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.104930] Serial: 8250/16550 driver, 2 ports, IRQ sharing enabled
[    0.105535] printk: console [ttyS0] disabled
[    0.105757] serial8250.0: ttyS0 at MMIO 0xffe04500 (irq = 42, base_baud = 24999999) is a 16550A
[    0.105801] printk: console [ttyS0] enabled
[    0.998473] printk: console [ttyS0] disabled
[    1.008282] ffe04500.serial: ttyS0 at MMIO 0xffe04500 (irq = 42, base_baud = 24999999) is a 16550
[    1.008320] printk: console [ttyS0] enabled
[    1.919107] fsl_espi ffe07000.spi: cs=0, init_csmode=0x100008
[    1.928741] spi-nor spi0.0: s25fl128s1 (16384 Kbytes)
[    1.934151] 5 fixed-partitions partitions found on MTD device spi0.0
[    1.940563] OF: Bad cell count for /soc@ffe00000/spi@7000/flash@0/partitions
[    1.947629] OF: Bad cell count for /soc@ffe00000/spi@7000/flash@0/partitions
[    1.954854] Creating 5 MTD partitions on "spi0.0":
[    1.959660] 0x000000000000-0x000000050000 : "u-boot"
[    1.966329] OF: Bad cell count for /soc@ffe00000/spi@7000/flash@0/partitions
[    1.973642] 0x000000050000-0x000000060000 : "dtb"
[    1.979585] 0x000000060000-0x000000fe0000 : "firmware"
[    1.985977] 2 tplink-fw partitions found on MTD device firmware
[    1.991941] Creating 2 MTD partitions on "firmware":
[    1.996904] 0x000000000000-0x000000350ad4 : "kernel"
[    2.001871] mtd: partition "kernel" doesn't end on an erase/write block -- force read-only
[    2.011249] 0x000000350ad4-0x000000f80000 : "rootfs"
[    2.016227] mtd: partition "rootfs" doesn't start on an erase/write block boundary -- force read-only
[    2.026438] mtd: setting mtd4 (rootfs) as root device
[    2.031594] 1 squashfs-split partitions found on MTD device rootfs
[    2.037779] 0x000000710000-0x000000f80000 : "rootfs_data"
[    2.044231] 0x000000fe0000-0x000000ff0000 : "config"
[    2.050365] 0x000000ff0000-0x000001000000 : "caldata"
[    2.056534] OF: Bad cell count for /soc@ffe00000/spi@7000/flash@0/partitions
[    2.066200] fsl_espi ffe07000.spi: irq = 59
[    2.094523] fsl-gianfar soc@ffe00000:ethernet@b0000: enabled errata workarounds, flags: 0x4
[    2.113186] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: mac: 64:66:b3:de:ef:b5
[    2.120614] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: Running with NAPI enabled
[    2.128268] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: RX BD ring size for Q[0]: 256
[    2.136276] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: TX BD ring size for Q[0]: 256
[    2.144483] ucc_geth_driver: QE UCC Gigabit Ethernet Controller
[    2.150854] mpc-i2c ffe03000.i2c: timeout 1000000 us
[    2.155986] mpc-i2c ffe03100.i2c: timeout 1000000 us
[    2.161193] booke_wdt: powerpc book-e watchdog driver loaded
[    2.166995] Freescale(R) MPC85xx EDAC driver, (C) 2006 Montavista Software
[    2.174540] EDAC PCI0: Giving out device to module MPC85xx_edac controller mpc85xx_pci_err: DEV mpc85xx-pci-edac.0.auto (INTERRUPT)
[    2.186424] MPC85xx_edac acquired irq 16 for PCI Err
[    2.191402] MPC85xx_edac PCI err registered
[    2.195653] EDAC PCI1: Giving out device to module MPC85xx_edac controller mpc85xx_pci_err: DEV mpc85xx-pci-edac.1.auto (INTERRUPT)
[    2.207502] MPC85xx_edac acquired irq 16 for PCI Err
[    2.212471] MPC85xx_edac PCI err registered
[    2.217666] NET: Registered PF_INET6 protocol family
[    2.227527] Segment Routing with IPv6
[    2.231292] In-situ OAM (IOAM) with IPv6
[    2.235315] NET: Registered PF_PACKET protocol family
[    2.240566] 8021q: 802.1Q VLAN Support v1.8
[    2.244807] drmem: No dynamic reconfiguration memory found
[    2.336559] qca8k mdio@ffe24000:10: configuring for fixed/rgmii-id link mode
[    2.346471] qca8k mdio@ffe24000:10: Link is Up - 1Gbps/Full - flow control off
[    2.429778] qca8k mdio@ffe24000:10 wan (uninitialized): PHY [mdio@ffe24000:00] driver [Qualcomm Atheros 8327-B internal PHY] (irq=POLL)
[    2.519790] qca8k mdio@ffe24000:10 lan1 (uninitialized): PHY [mdio@ffe24000:01] driver [Qualcomm Atheros 8327-B internal PHY] (irq=POLL)
[    2.609787] qca8k mdio@ffe24000:10 lan2 (uninitialized): PHY [mdio@ffe24000:02] driver [Qualcomm Atheros 8327-B internal PHY] (irq=POLL)
[    2.699817] qca8k mdio@ffe24000:10 lan3 (uninitialized): PHY [mdio@ffe24000:03] driver [Qualcomm Atheros 8327-B internal PHY] (irq=POLL)
[    2.789788] qca8k mdio@ffe24000:10 lan4 (uninitialized): PHY [mdio@ffe24000:04] driver [Qualcomm Atheros 8327-B internal PHY] (irq=POLL)
[    2.808061] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: entered promiscuous mode
[    2.815735] DSA: tree 0 setup
[    2.824921] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    2.832573] Freeing unused kernel image (initmem) memory: 192K
[    2.838427] Run /sbin/init as init process
[    2.842540]   with arguments:
[    2.842544]     /sbin/init
[    2.842549]   with environment:
[    2.842552]     HOME=/
[    2.842557]     TERM=linux
[    3.028560] random: crng init done
[    3.247310] init: Console is alive
[    3.250984] init: - watchdog -
[    4.102082] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.160197] usbcore: registered new interface driver usbfs
[    4.165741] usbcore: registered new interface driver hub
[    4.171146] usbcore: registered new device driver usb
[    4.181670] button_hotplug: loading out-of-tree module taints kernel.
[    4.188620] Button Hotplug driver version 0.4.1
[    4.200671] fsl-ehci fsl-ehci.0: Freescale On-Chip EHCI Host Controller
[    4.207321] fsl-ehci fsl-ehci.0: new USB bus registered, assigned bus number 1
[    4.224452] fsl-ehci fsl-ehci.0: irq 28, io mem 0xffe22000
[    4.258512] fsl-ehci fsl-ehci.0: USB 2.0 started, EHCI 1.00
[    4.264902] hub 1-0:1.0: USB hub found
[    4.268939] hub 1-0:1.0: 1 port detected
[    4.279824] input: buttons as /devices/platform/buttons/input/input0
[    4.287526] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.297599] init: - preinit -
[    4.558549] usb 1-1: new high-speed USB device number 2 using fsl-ehci
[    4.791243] hub 1-1:1.0: USB hub found
[    4.795830] hub 1-1:1.0: 4 ports detected
[    5.004829] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: Link is Up - 1Gbps/Full - flow control off
[    7.175599] jffs2_scan_eraseblock(): End of filesystem marker found at 0x10000
[    7.182867] jffs2_build_filesystem(): unlocking the mtd device...
[    7.182874] done.
[    7.190966] jffs2_build_filesystem(): erasing all blocks after the end marker...
[   27.200464] done.
[   27.209896] jffs2: notice: (411) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   27.227056] mount_root: overlay filesystem has not been fully initialized yet
[   27.238099] mount_root: switching to jffs2 overlay
[   27.247694] overlayfs: upper fs does not support tmpfile.
[   27.694885] urandom-seed: Seed file not found (/etc/urandom.seed)
[   27.780661] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: Link is Down
[   27.797881] procd: - early -
[   27.800966] procd: - watchdog -
[   28.418245] procd: - watchdog -
[   28.421701] procd: - ubus -
[   28.489428] procd: - init -
[   29.240021] kmodloader: loading kernel modules from /etc/modules.d/*
[   29.734262] Loading modules backported from Linux version v6.9.9-0-g28fdf4518483
[   29.741704] Backport generated by backports.git v6.1.97-1-29-gf1d24a3683b2
[   30.251670] PPP generic driver version 2.4.2
[   30.270162] NET: Registered PF_PPPOX protocol family
[   30.284184] wireguard: WireGuard 1.0.0 loaded. See www.wireguard.com for information.
[   30.292054] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[   30.387569] ath9k 9000:01:00.0: Direct firmware load for pci_wmac0.eeprom failed with error -2
[   30.396231] ath9k 9000:01:00.0: Falling back to sysfs fallback for: pci_wmac0.eeprom
[   30.508504] ath: phy0: Unable to load EEPROM file pci_wmac0.eeprom
[   30.514710] ath9k 9000:01:00.0: Failed to initialize device
[   30.520321] ath9k: probe of 9000:01:00.0 failed with error -22
[   30.568614] ath9k a000:03:00.0: Direct firmware load for pci_wmac1.eeprom failed with error -2
[   30.577238] ath9k a000:03:00.0: Falling back to sysfs fallback for: pci_wmac1.eeprom
[   30.668728] ath: phy1: Unable to load EEPROM file pci_wmac1.eeprom
[   30.674931] ath9k a000:03:00.0: Failed to initialize device
[   30.680556] ath9k: probe of a000:03:00.0 failed with error -22
[   30.718929] kmodloader: done loading kernel modules from /etc/modules.d/*
[   31.036962] urngd: v1.0.2 started.
[   40.293120] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: Link is Up - 1Gbps/Full - flow control off
[   40.312930] qca8k mdio@ffe24000:10 lan1: configuring for phy/gmii link mode
[   40.345526] br-lan: port 1(lan1) entered blocking state
[   40.350811] br-lan: port 1(lan1) entered disabled state
[   40.356073] qca8k mdio@ffe24000:10 lan1: entered allmulticast mode
[   40.362279] fsl-gianfar soc@ffe00000:ethernet@b0000 eth0: entered allmulticast mode
[   40.386142] qca8k mdio@ffe24000:10 lan1: entered promiscuous mode
[   40.417042] qca8k mdio@ffe24000:10 lan2: configuring for phy/gmii link mode
[   40.434926] br-lan: port 2(lan2) entered blocking state
[   40.440206] br-lan: port 2(lan2) entered disabled state
[   40.445466] qca8k mdio@ffe24000:10 lan2: entered allmulticast mode
[   40.459293] qca8k mdio@ffe24000:10 lan2: entered promiscuous mode
[   40.488919] qca8k mdio@ffe24000:10 lan3: configuring for phy/gmii link mode
[   40.508917] br-lan: port 3(lan3) entered blocking state
[   40.514165] br-lan: port 3(lan3) entered disabled state
[   40.519452] qca8k mdio@ffe24000:10 lan3: entered allmulticast mode
[   40.533480] qca8k mdio@ffe24000:10 lan3: entered promiscuous mode
[   40.553735] qca8k mdio@ffe24000:10 lan4: configuring for phy/gmii link mode
[   40.564864] br-lan: port 4(lan4) entered blocking state
[   40.570151] br-lan: port 4(lan4) entered disabled state
[   40.575414] qca8k mdio@ffe24000:10 lan4: entered allmulticast mode
[   40.588419] qca8k mdio@ffe24000:10 lan4: entered promiscuous mode
[   40.618790] qca8k mdio@ffe24000:10 wan: configuring for phy/gmii link mode
[   44.890232] qca8k mdio@ffe24000:10 wan: Link is Up - 1Gbps/Full - flow control off
```

WAN has wrong ip:

```
root@OpenWrt:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1502 qdisc fq_codel state UP qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::6666:b3ff:fede:efb5/64 scope link
       valid_lft forever preferred_lft forever
3: wan@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 26:f5:a2:25:26:d3 brd ff:ff:ff:ff:ff:ff
    inet 192.168.11.10/24 brd 192.168.111.255 scope global wan
       valid_lft forever preferred_lft forever
    inet6 fe80::24f5:a2ff:fe25:26d3/64 scope link
       valid_lft forever preferred_lft forever
4: lan1@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
5: lan2@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
6: lan3@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
7: lan4@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
8: br-lan: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN qlen 1000
    link/ether 64:66:b3:de:ef:b5 brd ff:ff:ff:ff:ff:ff
    inet 192.168.70.1/24 brd 192.168.70.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd10:ec20:5d7e::1/60 scope global tentative noprefixroute
       valid_lft forever preferred_lft forever
```

CHKDSK88 commented:
```
root@OpenWrt:~# lspci -v
9000:00:00.0 PCI bridge: Freescale Semiconductor Inc Device 012b (rev 10) (prog-if 00 [Normal decode])
        Device tree node: /sys/firmware/devicetree/base/pcie@ffe09000/pcie@0
        Flags: bus master, fast devsel, latency 0, IRQ 16
        Bus: primary=00, secondary=01, subordinate=01, sec-latency=0
        I/O behind bridge: 0000-ffff [size=64K] [16-bit]
        Memory behind bridge: a0000000-bfffffff [size=512M] [32-bit]
        Prefetchable memory behind bridge: [disabled] [64-bit]
        Capabilities: [44] Power Management version 3
        Capabilities: [4c] Express Root Port (Slot-), IntMsgNum 0
        Capabilities: [100] Advanced Error Reporting
        Kernel driver in use: pcieport
lspci: Unable to load libkmod resources: error -2

9000:01:00.0 Network controller: Qualcomm Atheros AR958x 802.11abgn Wireless Network Adapter (rev 01)
        Subsystem: Qualcomm Atheros AR9582 802.11a/n WLAN Mini-PCIe Adapter
        Device tree node: /sys/firmware/devicetree/base/pcie@ffe09000/pcie@0/wifi@0,0
        Flags: fast devsel, IRQ 17
        Memory at a0000000 (64-bit, non-prefetchable) [size=128K]
        Expansion ROM at a0020000 [virtual] [disabled] [size=64K]
        Capabilities: [40] Power Management version 3
        Capabilities: [50] MSI: Enable- Count=1/4 Maskable+ 64bit+
        Capabilities: [70] Express Endpoint, IntMsgNum 0
        Capabilities: [100] Advanced Error Reporting
        Capabilities: [140] Virtual Channel
        Capabilities: [300] Device Serial Number 00-00-00-00-00-00-00-00

a000:02:00.0 PCI bridge: Freescale Semiconductor Inc Device 012b (rev 10) (prog-if 00 [Normal decode])
        Device tree node: /sys/firmware/devicetree/base/pcie@ffe0a000/pcie@0
        Flags: bus master, fast devsel, latency 0, IRQ 16
        Bus: primary=00, secondary=03, subordinate=03, sec-latency=0
        I/O behind bridge: 0000-ffff [size=64K] [16-bit]
        Memory behind bridge: 80000000-9fffffff [size=512M] [32-bit]
        Prefetchable memory behind bridge: [disabled] [64-bit]
        Capabilities: [44] Power Management version 3
        Capabilities: [4c] Express Root Port (Slot-), IntMsgNum 0
        Capabilities: [100] Advanced Error Reporting
        Kernel driver in use: pcieport

a000:03:00.0 Network controller: Qualcomm Atheros AR93xx Wireless Network Adapter (rev 01)
        Device tree node: /sys/firmware/devicetree/base/pcie@ffe0a000/pcie@0/wifi@0,0
        Flags: fast devsel, IRQ 18
        Memory at 80000000 (64-bit, non-prefetchable) [size=128K]
        Expansion ROM at 80020000 [virtual] [disabled] [size=64K]
        Capabilities: [40] Power Management version 3
        Capabilities: [50] MSI: Enable- Count=1/4 Maskable+ 64bit+
        Capabilities: [70] Express Endpoint, IntMsgNum 0
        Capabilities: [100] Advanced Error Reporting
        Capabilities: [140] Virtual Channel
        Capabilities: [300] Device Serial Number 00-00-00-00-00-00-00-00

```

neheb commented:
I think it has something to do with those PCI remapping warnings. Unfortunately it looks like I’ll have to drop that commit as I don’t own the hardware for further testing.

neheb commented:
Rebased.

@DragonBluep is CONFIG_NVMEM_SYSFS needed here?

CHKDSK88 commented:
EEPROM content:

```
root@OpenWrt:~# hexdump -C /dev/mtd7 
00000000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00001000  02 06 00 02 03 04 05 06  00 30 3a 30 32 3a 30 33  |.........0:02:03|
00001010  3a 30 34 3a 30 35 3a 30  36 00 00 00 00 00 1f 00  |:04:05:06.......|
00001020  77 01 00 00 00 00 04 00  48 00 4d 04 06 00 08 ff  |w.......H.M.....|
00001030  30 00 00 00 00 10 01 00  00 44 44 04 00 50 01 50  |0........DD..P.P|
00001040  01 50 01 00 00 00 00 00  00 44 00 a4 00 00 00 00  |.P.......D......|
00001050  ff 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00001060  0e 0e 03 00 2c e2 00 02  0e 1c 80 c0 80 0c 80 c0  |....,...........|
00001070  80 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00001080  00 00 00 00 00 00 00 00  de de 70 89 a2 00 00 00  |..........p.....|
00001090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
000010c0  00 00 00 70 ac 70 89 ac  70 89 ac 70 89 ac 22 22  |...p.p..p..p..""|
000010d0  22 22 22 22 22 22 22 22  20 20 22 22 20 20 22 22  |""""""""  ""  ""|
000010e0  20 20 20 20 20 20 20 1c  20 20 1e 1c 00 00 00 00  |       .  ......|
000010f0  20 20 20 20 20 1c 20 20  1e 1c 00 00 00 00 20 20  |     .  ......  |
00001100  20 20 20 1c 20 20 1e 1c  00 00 00 00 1e 1e 1e 1e  |   .  ..........|
00001110  1e 1c 1e 1e 1c 1a 00 00  00 00 1e 1e 1e 1e 1e 1c  |................|
00001120  1e 1e 1c 1a 00 00 00 00  1e 1e 1e 1e 1e 1c 1e 1e  |................|
00001130  1c 1a 00 00 00 00 11 12  15 17 41 42 45 47 31 32  |..........ABEG12|
00001140  35 37 70 75 9d a2 70 75  a2 ff 70 75 a2 ff 7a 7f  |57pu..pu..pu..z.|
00001150  93 98 70 75 ac b8 70 75  ac 00 70 75 ac 00 7a 7f  |..pu..pu..pu..z.|
00001160  93 a2 70 75 ac 00 70 75  ac 00 70 75 ac 00 7a 7f  |..pu..pu..pu..z.|
00001170  93 a2 3c 7c 3c 3c 3c 7c  3c 3c 7c 3c 3c 7c 7c 3c  |..<|<<<|<<|<<||<|
00001180  00 00 3c 7c 3c 3c 3c 7c  3c 3c 3c 7c 7c 3c 3c 7c  |..<|<<<|<<<||<<||
00001190  3c 3c 3c 7c 3c 3c 3c 7c  3c 3c 3c 7c 7c 7c 3c 7c  |<<<|<<<|<<<|||<||
000011a0  7c 7c 10 01 00 00 22 22  02 00 10 00 10 00 10 00  |||....""........|
000011b0  00 00 00 00 00 00 44 00  00 00 00 00 00 ff 00 00  |......D.........|
000011c0  00 00 00 00 00 00 00 00  00 00 00 de 0f 0e 0e 03  |................|
000011d0  00 2d e2 00 02 0e 1c e0  e0 f0 0c e0 e0 f0 6c 00  |.-............l.|
000011e0  00 00 00 00 00 00 00 00  00 32 44 00 00 00 00 00  |.........2D.....|
000011f0  00 00 00 00 00 00 00 4c  54 6c 7c 8c a4 bd cd f1  |.......LTl|.....|
00001200  00 a5 00 00 00 f4 00 a5  00 00 00 f7 00 a6 00 00  |................|
00001210  00 fb 00 a5 00 00 00 ff  00 a6 00 00 00 fe 00 a6  |................|
00001220  00 00 00 f6 00 a6 00 00  00 f3 00 a5 00 00 00 ed  |................|
00001230  00 a5 00 00 00 f1 00 a5  00 00 00 f5 00 a6 00 00  |................|
00001240  00 f7 00 a6 00 00 00 fa  00 a7 00 00 00 f9 00 a6  |................|
00001250  00 00 00 f2 00 a7 00 00  00 ed 00 a6 00 00 00 f3  |................|
00001260  00 a5 00 00 00 f6 00 a6  00 00 00 fc 00 a6 00 00  |................|
00001270  00 00 00 a6 00 00 00 02  00 a6 00 00 00 00 00 a6  |................|
00001280  00 00 00 fb 00 a6 00 00  00 f7 00 a7 00 00 00 4c  |...............L|
00001290  58 68 78 8c a0 b4 cd 4c  58 68 78 8c a0 b4 cd 4c  |Xhx....LXhx....L|
000012a0  58 68 78 8c a0 b4 cd 22  22 20 1a 22 22 20 1a 22  |Xhx...."" ."" ."|
000012b0  22 20 1a 22 22 20 1a 22  22 20 1a 22 22 20 1a 22  |" ."" ."" ."" ."|
000012c0  22 20 1a 22 22 20 1a 22  22 22 20 1e 18 22 1e 1a  |" ."" .""" .."..|
000012d0  18 1a 1a 1a 16 22 22 22  20 1e 18 22 1e 1a 18 1a  |.....""" .."....|
000012e0  1a 1a 16 22 22 22 20 1e  18 22 1e 1a 18 1a 1a 1a  |...""" .."......|
000012f0  16 22 22 22 20 1e 18 22  1e 1a 18 1a 1a 1a 16 22  |.""" .."......."|
00001300  22 22 20 1e 18 22 1e 1a  18 1a 1a 1a 16 22 22 22  |"" .."......."""|
00001310  20 1e 18 22 1e 1a 18 1a  1a 1a 16 22 22 22 20 1e  | ..".......""" .|
00001320  18 22 1e 1a 18 1a 1a 1a  16 22 22 22 20 1e 18 22  |.".......""" .."|
00001330  1e 1a 18 1a 1a 1a 16 22  22 20 1e 1e 18 22 1e 18  |......."" ..."..|
00001340  18 1c 1a 18 16 22 22 20  1e 1e 18 22 1e 18 18 1c  |....."" ..."....|
00001350  1a 18 16 22 22 20 1e 1e  18 22 1e 18 18 1c 1a 18  |..."" ..."......|
00001360  16 22 22 20 1e 1e 18 22  1e 18 18 1c 1a 18 16 22  |."" ..."......."|
00001370  22 20 1e 1e 18 22 1e 18  18 1c 1a 18 16 22 22 20  |" ..."......."" |
00001380  1e 1e 18 22 1e 18 18 1c  1a 18 16 22 22 20 1e 1e  |..."......."" ..|
00001390  18 22 1e 18 18 1c 1a 18  16 22 22 20 1e 1e 18 22  |."......."" ..."|
000013a0  1e 18 18 1c 1a 18 16 10  16 18 40 46 48 30 36 38  |..........@FH068|
000013b0  4c 50 68 78 8c a0 b9 cd  4c 50 68 78 8c a0 b9 cd  |LPhx....LPhx....|
000013c0  4c 50 68 78 8c a0 b9 cd  4c 54 68 78 8c a0 b9 cd  |LPhx....LThx....|
000013d0  4c 54 68 78 8c a0 b9 cd  4c 54 68 78 8c a0 b9 cd  |LThx....LThx....|
000013e0  4c 50 68 78 8c a0 b9 cd  4c 50 68 78 8c a0 b9 cd  |LPhx....LPhx....|
000013f0  4c 50 68 78 8c a0 b9 cd  62 66 66 66 66 66 66 26  |LPhx....bffffff&|
00001400  5d 66 66 66 66 66 66 26  54 66 66 66 66 66 66 26  |]ffffff&Tffffff&|
00001410  3c 7c 3c 7c 3c 7c 3c 7c  3c 7c 3c 7c 3c 7c 3c 7c  |<|<|<|<|<|<|<|<||
00001420  3c 7c 3c 7c 3c 7c 3c 7c  56 56 56 56 56 56 56 16  |<|<|<|<|VVVVVVV.|
00001430  56 56 56 56 56 56 56 16  56 56 56 56 56 56 56 16  |VVVVVVV.VVVVVVV.|
00001440  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00005000  02 02 00 02 03 04 05 06  00 00 00 00 00 00 00 00  |................|
00005010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 1f 00  |................|
00005020  77 02 00 00 00 00 04 00  48 00 4d 04 03 00 08 ff  |w.......H.M.....|
00005030  00 00 00 00 00 10 01 00  00 22 22 02 00 10 00 10  |........."".....|
00005040  00 10 00 00 00 00 00 00  00 21 00 a4 00 00 00 00  |.........!......|
00005050  ff 00 00 00 00 00 00 00  00 00 00 00 00 00 00 0f  |................|
00005060  0e 0e 03 00 2c e2 00 02  0e 1c 80 c0 80 0c 80 c0  |....,...........|
00005070  80 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00005080  00 00 00 00 00 00 00 00  00 00 70 89 a2 0c 00 95  |..........p.....|
00005090  00 00 00 0e 00 96 00 00  00 10 00 97 00 00 00 11  |................|
000050a0  00 95 00 00 00 12 00 95  00 00 00 11 00 95 00 00  |................|
000050b0  00 10 00 96 00 00 00 0f  00 97 00 00 00 0d 00 97  |................|
000050c0  00 00 00 70 ac 70 89 ac  70 89 ac 7a 89 ac 28 28  |...p.p..p..z..((|
000050d0  28 28 28 28 28 28 28 28  26 22 28 28 26 22 28 28  |((((((((&"((&"((|
000050e0  26 22 26 26 26 24 22 20  26 22 22 20 26 22 22 1e  |&"&&&$" &"" &"".|
000050f0  26 26 26 24 22 20 26 22  22 20 26 22 22 1e 26 26  |&&&$" &"" &"".&&|
00005100  26 24 22 20 26 22 22 20  26 22 22 1e 24 24 24 22  |&$" &"" &"".$$$"|
00005110  20 1e 24 22 20 1c 24 22  20 1c 24 24 24 22 20 1e  | .$" .$" .$$$" .|
00005120  24 22 20 1c 24 22 20 1c  24 24 24 22 20 1e 24 22  |$" .$" .$$$" .$"|
00005130  20 1c 24 22 20 1c 11 12  15 17 41 42 45 47 31 32  | .$" .....ABEG12|
00005140  35 37 70 75 7a a2 70 75  7a a2 70 75 7a a2 7a 7f  |57puz.puz.puz.z.|
00005150  84 98 70 75 ac b8 70 75  ac b8 70 75 ac b8 70 75  |..pu..pu..pu..pu|
00005160  ac b8 70 75 7a ac 70 75  7a ac 70 75 7a ac 7a 75  |..puz.puz.puz.zu|
00005170  7a a2 65 6a 6a 28 5a 6a  6a 1f 58 68 68 1e 54 7c  |z.ejj(Zjj.Xhh.T||
00005180  7c 19 3c 7c 3c 7c 3c 7c  3c 7c 3c 7c 3c 7c 3c 7c  ||.<|<|<|<|<|<|<||
00005190  3c 7c 5a 5a 5a 1a 5a 5a  5a 1a 5a 5a 5a 1a 5a 5a  |<|ZZZ.ZZZ.ZZZ.ZZ|
000051a0  5a 1a 10 01 00 00 22 22  02 00 00 00 00 00 00 00  |Z.....""........|
000051b0  00 00 00 00 00 00 44 00  00 00 00 00 00 ff 00 00  |......D.........|
000051c0  00 00 00 00 00 00 00 00  00 00 00 00 00 0e 0e 03  |................|
000051d0  00 2d e2 00 02 0e 1c e0  e0 f0 0c e0 e0 f0 6c 00  |.-............l.|
000051e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000051f0  00 00 00 00 00 00 00 4c  54 68 78 8c a0 b9 cd 00  |.......LThx.....|
00005200  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00005280  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 4c  |...............L|
00005290  54 68 78 8c a0 b9 cd 4c  58 68 8c b4 bd b9 cd 4c  |Thx....LXh.....L|
000052a0  58 68 8c b4 bd b9 cd 14  14 14 0a 14 14 14 0a 14  |Xh..............|
000052b0  14 14 0a 14 14 14 0a 14  14 14 0a 14 14 14 0a 14  |................|
000052c0  14 14 0a 14 14 14 0a 14  14 0a 0a 00 00 0a 0a 00  |................|
000052d0  00 0a 0a 00 00 14 14 0a  0a 00 00 0a 0a 00 00 0a  |................|
000052e0  0a 00 00 14 14 0a 0a 00  00 0a 0a 00 00 0a 0a 00  |................|
000052f0  00 14 14 0a 0a 00 00 0a  0a 00 00 0a 0a 00 00 14  |................|
00005300  14 0a 0a 00 00 0a 0a 00  00 0a 0a 00 00 14 14 0a  |................|
00005310  0a 00 00 0a 0a 00 00 0a  0a 00 00 14 14 0a 0a 00  |................|
00005320  00 0a 0a 00 00 0a 0a 00  00 14 14 0a 0a 00 00 0a  |................|
00005330  0a 00 00 0a 0a 00 00 14  14 0a 0a 00 00 0a 0a 00  |................|
00005340  00 0a 0a 00 00 14 14 0a  0a 00 00 0a 0a 00 00 0a  |................|
00005350  0a 00 00 14 14 0a 0a 00  00 0a 0a 00 00 0a 0a 00  |................|
00005360  00 14 14 0a 0a 00 00 0a  0a 00 00 0a 0a 00 00 14  |................|
00005370  14 0a 0a 00 00 0a 0a 00  00 0a 0a 00 00 14 14 0a  |................|
00005380  0a 00 00 0a 0a 00 00 0a  0a 00 00 14 14 0a 0a 00  |................|
00005390  00 0a 0a 00 00 0a 0a 00  00 14 14 0a 0a 00 00 0a  |................|
000053a0  0a 00 00 0a 0a 00 00 10  16 18 40 46 48 30 36 38  |..........@FH068|
000053b0  4c 5c 60 8c a0 b4 bd cd  4c 5c 60 8c 90 b4 bd cd  |L\`.....L\`.....|
000053c0  4e 56 5e 66 8e 96 ae bf  4c 50 5c 68 8c b4 ff ff  |NV^f....LP\h....|
000053d0  4c 5c 8c b4 ff ff ff ff  4e 5e 66 8e 9e ae ff ff  |L\......N^f.....|
000053e0  4c 50 54 5c 8c a0 b4 bd  4c 5c 68 8c 98 b4 bd cd  |LPT\....L\h.....|
000053f0  4e 56 5e 8e 96 ae bf c7  7c 7c 7c 7c 7c 7c 7c 3c  |NV^.....|||||||<|
00005400  7c 7c 7c 7c 7c 7c 7c 3c  3c 7c 3c 7c 7c 7c 7c 7c  ||||||||<<|<||||||
00005410  3c 7c 7c 3c 7c 3c 3c 3c  7c 7c 7c 3c 3c 3c 3c 3c  |<||<|<<<|||<<<<<|
00005420  7c 7c 7c 7c 7c 3c 3c 3c  7c 7c 7c 7c 7c 7c 7c 7c  ||||||<<<|||||||||
00005430  7c 7c 3c 7c 7c 7c 7c 3c  7c 3c 7c 7c 7c 7c 3c 7c  |||<||||<|<||||<||
00005440  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00010000
root@OpenWrt:~# 
```

neheb commented:
Ok added, CONFIG_NVMEM_SYSFS=y. Based on those ath9k errors, it seems it's needed.

neheb commented:
Why does this platform use C files?

neheb committed:
mpc85xx: tl-wdr4900: move platform code to dts

No benefit in doing so in platform file. Code for dts has already been
written. Might as well use it.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16125
Signed-off-by: Robert Marko <robimarko@gmail.com>

CHKDSK88 commented on target/linux/mpc85xx/config-6.6:
```diff
@@ -178,6 +178,7 @@ CONFIG_NR_IRQS=512
 CONFIG_NVMEM=y
 CONFIG_NVMEM_LAYOUTS=y
 # CONFIG_NVMEM_QORIQ_EFUSE is not set
+CONFIG_NVMEM_SYSFS=y
```
You can drop it. It works without it.

CHKDSK88 commented:
`Tested-by: Pawel Dembicki <paweldembicki@gmail.com>`

Now everything works:

```
[   30.504182] ath: EEPROM regdomain: 0x0
[   30.504204] ath: EEPROM indicates default country code should be used
[   30.504208] ath: doing EEPROM country->regdmn map search
[   30.504218] ath: country maps to regdmn code: 0x3a
[   30.504224] ath: Country alpha2 being used: US
[   30.504229] ath: Regpair used: 0x3a
[   30.512520] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   30.513266] ieee80211 phy0: Atheros AR9300 Rev:4 mem=0xe758a2b1, irq=17
[   30.573832] ath: EEPROM regdomain: 0x0
[   30.573856] ath: EEPROM indicates default country code should be used
[   30.573860] ath: doing EEPROM country->regdmn map search
[   30.573870] ath: country maps to regdmn code: 0x3a
[   30.573876] ath: Country alpha2 being used: US
[   30.573881] ath: Regpair used: 0x3a
[   30.582225] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   30.582916] ieee80211 phy1: Atheros AR9300 Rev:3 mem=0xdef4fae8, irq=18
```

neheb commented:
Pushed a new commit with moves to DTS.

My suspicion is that this will break wireless because of qca,led-pin. At least it does on ath79.

CHKDSK88 commented:
I can see `/sys/class/leds/tp-link:blue:wps/brightness` on system, but it doesn't work.

neheb commented:
Changed GPIO controller to 2ghz interface, as in the C file.

CHKDSK88 commented on target/linux/mpc85xx/files/arch/powerpc/boot/dts/tl-wdr4900-v1.dts:
```diff
@@ -271,6 +316,11 @@
 
+		wps {
+			gpios = <&ath9k 1 GPIO_ACTIVE_LOW>;
+			label = "tp-link:blue:wps";
```
nitpick: led is green

neheb replied:
Fixed.

neheb replied:
Although to be honest these should be using the new LED macros.

CHKDSK88 commented:
Everything work as expected.

robimarko commented:
Thanks! Rebased on top of main and merged!
